### PR TITLE
[grub]: use UUID for root partition

### PR DESCRIPTION
**Pull request by lguohan**
Signed-off-by: Guohan Lu <lguohan@gmail.com>

**- What I did**

**- How I did it**

**- How to verify it**
```

                             GNU GRUB  version 2.02

 +----------------------------------------------------------------------------+
 |        insmod part_msdos                                                   |^
 |        insmod ext2                                                         |
 |        linux   /image-uuid.0-dirty-20200418.115911/boot/vmlinuz-4.19.0-6-a\|
 |md64 root=UUID=0530e631-21fa-4707-bbf0-6100631556e3 rw console=tty0 console\|
 |=ttyS0,115200n8 quiet                  net.ifnames=0 biosdevname=0         \|
 |        loop=image-uuid.0-dirty-20200418.115911/fs.squashfs loopfstype=squa\|
 |shfs                                       apparmor=1 security=apparmor var\|
 |log_size=4096 usbcore.autosuspend=-1                                        |
 |        echo    'Loading SONiC-OS OS initial ramdisk ...'                   |
 |        initrd  /image-uuid.0-dirty-20200418.115911/boot/initrd.img-4.19.0-\|
 |6-amd64                                                                     |
 |                                                                            | 
 +----------------------------------------------------------------------------+

```
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: installer/x86_64/install.sh [hunk 1]
```diff
@@ -583,19 +583,20 @@ EOF
...
+    grub_cfg_root=$(echo $old_sonic_menuentry | sed -e "s/.*root\=\(.*\)rw.*/\1/")
     onie_menuentry=$(cat /host/grub/grub.cfg | sed "/menuentry ONIE/,/}/!d")
-fi
-
-if [ "$install_env" = "build" ]; then
+elif [ "$install_env" = "build" ]; then
     grub_cfg_root=%%SONIC_ROOT%%
 else
-    grub_cfg_root=$demo_dev
+    grub_cfg_root=$(blkid "$demo_dev" | grep -oE "[[:blank:]]UUID=\".*\"" | sed s/\"//g | sed s/\ //g)
```
Comment (qiluo-msft) [inline installer/x86_64/install.sh:596]: >blkid "$demo_dev" | grep -oE "[[:blank:]]UUID=\".*\"" | sed s/\"//g | sed s/\ //g [](start = 20, length = 81)

```suggestion
    grub_cfg_root=$(blkid "$demo_dev" | sed 's/.*UUID=\"\([^"]*\)\".*/\1/g')
```

File: installer/x86_64/install.sh [hunk 2]
```diff
@@ -583,19 +583,20 @@ EOF
...
     old_sonic_menuentry=$(cat /host/grub/grub.cfg | sed "/$running_sonic_revision/,/}/!d")
-    demo_dev=$(echo $old_sonic_menuentry | sed -e "s/.*root\=\(.*\)rw.*/\1/")
+    grub_cfg_root=$(echo $old_sonic_menuentry | sed -e "s/.*root\=\(.*\)rw.*/\1/")
     onie_menuentry=$(cat /host/grub/grub.cfg | sed "/menuentry ONIE/,/}/!d")
-fi
-
-if [ "$install_env" = "build" ]; then
+elif [ "$install_env" = "build" ]; then
     grub_cfg_root=%%SONIC_ROOT%%
 else
```
Comment (qiluo-msft) [inline installer/x86_64/install.sh:595]: What is the `$install_env` in `else`?

Comment (lguohan) [reply installer/x86_64/install.sh:595]: added

File: installer/x86_64/install.sh (hunk 1 above)
Comment (lguohan) [reply installer/x86_64/install.sh:596]: the problem of the suggestion is that if there is UUID entry in the output, you will the get whole string. test following.

```
/dev/vda3: LABEL="SONiC-OS" PARTUUID="abc" DUUID="0530e631-21fa-4707-bbf0-6100631556e3" PARTUUID="xxx"
/dev/vda3: LABEL="SONiC-OS" PARTUUID="abc"
/dev/vda3: LABEL="SONiC-OS"
```

Comment (qiluo-msft) [reply installer/x86_64/install.sh:596]: I see. I refine the suggestion. You may commit it if you feel it is good.

---
In reply to: [410710790](https://github.com/Azure/sonic-buildimage/pull/4443#discussion_r410710790) [](ancestors = 410710790)

Comment (lguohan) [reply installer/x86_64/install.sh]: thanks

**Pull request merged**